src/replica/replica_stub.cpp

```diff
@@ -2033,6 +2033,21 @@ void replica_stub::open_replica(const app_info &app,
     }
 
     if (rep == nullptr) {
+        // NOTICE: if dir a.b.pegasus does not exist, or .app-info does not exist, but the ballot > 0
+        // or the last_committed_decree > 0, start replica will fail
+        if ((req2 != nullptr) && (req2->info.is_stateful)) {
```

Comment:
Could you please try to add a unit test for it?

Reply:
@acelyc111 @hycdong hi, help me review again, thanks.